File: internal/cli/rm.go
package cli

import (
	"fmt"

	"github.com/spf13/cobra"
)

type rmOptions struct {
	repos      []string
	force      bool
	continueOn bool
}

func newRmCmd(d Deps) *cobra.Command {
	var opts rmOptions
	cmd := &cobra.Command{
		Use:   "rm <branch>",
		Short: "Remove worktrees for the branch across configured repos",
		Long: `Remove the git worktree for <branch> in each selected repo (serial).

Uses git worktree remove. Dirty / locked / residual worktrees require --force.

With --continue, remaining repos still run after a failure; the command
exits non-zero if any repo failed.`,
		Args: cobra.ExactArgs(1),
		RunE: func(cmd *cobra.Command, args []string) error {
			return runRm(d, args[0], opts)
		},
	}
	cmd.Flags().StringSliceVar(&opts.repos, "repos", nil, "subset of config repos (comma-separated or repeated)")
	cmd.Flags().BoolVar(&opts.force, "force", false, "pass --force to git worktree remove (dirty/locked/residual)")
	cmd.Flags().BoolVar(&opts.continueOn, "continue", false, "best-effort: continue after a repo failure; still exit non-zero if any failed")
	return cmd
}

func runRm(d Deps, branch string, opts rmOptions) error {
	if branch == "" {
		return fmt.Errorf("branch is required")
	}
	cfg, err := d.loadConfig()
	if err != nil {
		return err
	}
	repos, err := selectRepos(cfg, opts.repos)
	if err != nil {
		return err
	}
	if len(repos) == 0 {
		return fmt.Errorf("no repos selected")
	}

	out := d.stdout()
	return runSerial(repos, opts.continueOn, func(repo string) error {
		ctx, err := resolveRepo(cfg, repo, branch)
		if err != nil {
			return err
		}
		if err := d.git().Remove(ctx.MainPath, ctx.WorktreePath, opts.force); err != nil {
			return err
		}
		fmt.Fprintf(out, "%s\t%s\n", repo, ctx.WorktreePath)
		return nil
	})
}

File: src/cli/rm.rs
//! `mwt rm <branch>` — remove worktrees for the branch across configured repos.

use std::io::Write;

use clap::Args;

use crate::cli::deps::Deps;
use crate::cli::shared::{display_path, max_repo_width, resolve_repo, run_serial, select_repos};

#[derive(Debug, Args)]
pub struct RmOptions {
    #[arg(long, value_delimiter = ',')]
    pub repos: Vec<String>,

    /// Pass `--force` to `git worktree remove` (dirty/locked/residual).
    #[arg(long)]
    pub force: bool,

    /// Best-effort: continue after a repo failure; still exit non-zero if any failed.
    #[arg(long)]
    pub r#continue: bool,
}

pub fn run(deps: &Deps, branch: &str, opts: &RmOptions, out: &mut dyn Write) -> Result<(), String> {
    if branch.is_empty() {
        return Err("branch is required".to_string());
    }
    let cfg = (deps.load_config)().map_err(|e| format!("load config: {e}"))?;
    let repos = select_repos(&cfg, &opts.repos)?;
    if repos.is_empty() {
        return Err("no repos selected".to_string());
    }

    let repo_width = max_repo_width(&repos);

    run_serial(&repos, opts.r#continue, |repo| {
        let ctx = resolve_repo(&cfg, repo, branch)?;
        deps.git
            .remove(&ctx.main_path, &ctx.worktree_path, opts.force)?;
        writeln!(
            out,
            "{repo:<repo_width$}  {path}",
            path = display_path(&cfg.meta_root, &ctx.worktree_path.to_string_lossy())
        )
        .map_err(|e| e.to_string())?;
        Ok(())
    })
}
